recipe/build.sh

```diff
@@ -1,24 +1,20 @@
 #!/bin/bash
 
 if [[ $HOST == *apple* ]]; then
-    # Apply patch to build.py.in (autoreconf might not be needed?)
-    patch -p1 < ${RECIPE_DIR}/osx_patch.diff
-    # These two flags are mentioned in the LHAPDF docs
+    # Bug for macOS as of lhapdf v6.1.1+
+    # https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/INSTALL#L162-164
     export CFLAGS=-Qunused-arguments
     export CPPFLAGS=-Qunused-arguments
-fi
 
-if [[ $HOST == *arm64* ]]; then
-    # Get an updated config.sub and config.guess
-    cp $BUILD_PREFIX/share/gnuconfig/config.* ./config/
-    # And autoreconf!
-    autoreconf -i -f
-    (cd wrappers/python && cython --cplus lhapdf.pyx)
+    autoreconf --install --force
+    (cd wrappers/python && cython --include-dir . --cplus lhapdf.pyx)
 fi
 
-./configure --prefix=$PREFIX
+./configure --help
+
+./configure --prefix=$PREFIX CXXFLAGS="${CXXFLAGS}"
```

Comment:
Adding `CXXFLAGS` here has an effect on `lhapdf-config --cxxflags` but in testing on my M2 machine adding `LDFLAGS="${LDFLAGS}"` to the `configure` command as well didn't have any affect on `lhapdf-config --ldflags` so I gave up and went with adding `-Wl,-rpath,"$(lhapdf-config --libdir)"` to all compile steps by hand. (c.f. examples in `run_test.sh`).